spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -120,36 +111,39 @@ protected RewriteDataFilesSparkAction self() {
 
   @Override
   public RewriteDataFilesSparkAction binPack() {
-    Preconditions.checkArgument(
-        rewriter == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
-    this.rewriter = new SparkBinPackDataRewriter(spark(), table);
+    checkRunnerIsUnset();
+    this.runner = new SparkBinPackFileRewriteRunner(spark(), table);
     return this;
   }
 
   @Override
   public RewriteDataFilesSparkAction sort(SortOrder sortOrder) {
-    Preconditions.checkArgument(
-        rewriter == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
-    this.rewriter = new SparkSortDataRewriter(spark(), table, sortOrder);
+    checkRunnerIsUnset();
+    this.runner = new SparkSortFileRewriteRunner(spark(), table, sortOrder);
     return this;
   }
 
   @Override
   public RewriteDataFilesSparkAction sort() {
-    Preconditions.checkArgument(
-        rewriter == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
-    this.rewriter = new SparkSortDataRewriter(spark(), table);
+    checkRunnerIsUnset();
+    this.runner = new SparkSortFileRewriteRunner(spark(), table);
     return this;
   }
 
   @Override
   public RewriteDataFilesSparkAction zOrder(String... columnNames) {
-    Preconditions.checkArgument(
-        rewriter == null, "Must use only one rewriter type (bin-pack, sort, zorder)");
-    this.rewriter = new SparkZOrderDataRewriter(spark(), table, Arrays.asList(columnNames));
+    checkRunnerIsUnset();
+    this.runner = new SparkZOrderFileRewriteRunner(spark(), table, Arrays.asList(columnNames));
     return this;
   }
 
+  private void checkRunnerIsUnset() {
```

Comment:
nit: `ensureRunnerNotSet` is probably a little more clear

Reply:
Done